src/test/scala/group/scala/karazin/circe/literal/extras/PlainPrimitiveEncodeSuite.scala

```diff
@@ -1345,3 +1346,257 @@ class PlainPrimitiveEncodeSuite extends munit.ScalaCheckSuite:
+
+      val primitive = Primitive(monthDay)
+
+      val result: Json = JsonObject("value" -> monthDay.asJson).asJson
+
+      val expected: Json = encode"$primitive"
+
+      assertEquals(result, expected)
+    }
+
+  }
+
+  property("inlined primitive with OffsetTime value") {
+
+    case class Primitive(value: OffsetTime) derives Codec.AsObject
+
+    extension (inline sc: StringContext)
+      inline def encode(inline args: Any*): Json =
+        $ {macros.encode[Primitive]('sc, 'args)}
+
+    forAll { (offsetTime: OffsetTime) =>
+
+      val primitive = Primitive(offsetTime)
+
+      val result: Json = JsonObject("value" -> offsetTime.asJson).asJson
+
+      val expected: Json = encode"$primitive"
+
+      assertEquals(result, expected)
+    }
+
+  }
+
+  property("inlined primitive with OffsetDateTime value") {
+
+    case class Primitive(value: OffsetDateTime) derives Codec.AsObject
+
+    extension (inline sc: StringContext)
+      inline def encode(inline args: Any*): Json =
+        $ {macros.encode[Primitive]('sc, 'args)}
+
+    forAll { (offsetDateTime: OffsetDateTime) =>
+
+      val primitive = Primitive(offsetDateTime)
+
+      val result: Json = JsonObject("value" -> offsetDateTime.asJson).asJson
+
+      val expected: Json = encode"$primitive"
+
+      assertEquals(result, expected)
+    }
+
+  }
+
+
+
+  property("inlined primitive with YearMonth value") {
+
+    case class Primitive(value: YearMonth) derives Codec.AsObject
+
+    extension (inline sc: StringContext)
+      inline def encode(inline args: Any*): Json =
+        $ {macros.encode[Primitive]('sc, 'args)}
+
+    forAll { (yearMonth: YearMonth) =>
+
+      val primitive = Primitive(yearMonth)
+
+      val result: Json = JsonObject("value" -> yearMonth.asJson).asJson
+
+      val expected: Json = encode"$primitive"
+
+      assertEquals(result, expected)
+    }
+
+  }
+
+  property("inlined primitive with ZonedDateTime value") {
+
+    case class Primitive(value: ZonedDateTime) derives Codec.AsObject
+
+    extension (inline sc: StringContext)
+      inline def encode(inline args: Any*): Json =
+        $ {macros.encode[Primitive]('sc, 'args)}
+
+    forAll { (zonedDateTime: ZonedDateTime) =>
+
+      val primitive = Primitive(zonedDateTime)
+
+      val result: Json = JsonObject("value" -> zonedDateTime.asJson).asJson
+
+      val expected: Json = encode"$primitive"
+
+      assertEquals(result, expected)
+    }
+
+  }
+
+  property("inlined primitive with ZoneOffset value") {
+
+    case class Primitive(value: ZoneOffset) derives Codec.AsObject
+
+    extension (inline sc: StringContext)
+      inline def encode(inline args: Any*): Json =
+        $ {macros.encode[Primitive]('sc, 'args)}
+
+    forAll { (zoneOffset: ZoneOffset) =>
+
+      val primitive = Primitive(zoneOffset)
+
+      val result: Json = JsonObject("value" -> zoneOffset.asJson).asJson
+
+      val expected: Json = encode"$primitive"
+
+      assertEquals(result, expected)
+    }
+
+  }
```

Comment:
should leave an empty line at the eof
https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline/729795